src/middlewared/middlewared/plugins/boot_/format.py

```diff
@@ -0,0 +1,122 @@
+import platform
+
+from middlewared.schema import accepts, Dict, Int, Str
+from middlewared.service import CallError, private, Service
+from middlewared.utils import run
+
+IS_LINUX = platform.system().lower() == 'linux'
+
+
+class BootService(Service):
+
+    @accepts(
+        Str('dev'),
+        Dict(
+            'options',
+            Int('size'),
+            Int('swap_size'),
+        )
+    )
+    @private
+    async def format(self, dev, options):
+        """
+        Format a given disk `dev` using the appropriate partition layout
+        """
+        job = await self.middleware.call('disk.wipe', dev, 'QUICK')
+        await job.wait()
+        if job.error:
+            raise CallError(job.error)
+
+        disk_details = await self.middleware.call('device.get_disk', dev)
```

Comment:
I just noticed that device.get_disk returns a dict if the disk is not found?

Isn't that wrong/weird? Shouldnt it return None?